### PR TITLE
add progress bar for compression progress

### PR DESCRIPTION
**Pull request by jtarchie**
This only works for an input file because the sized of processed data can be determined. If the user uses STDING, a noop progress bar is used.

### CODE REVIEW

**Comment by jtarchie** [comment]
Related to #121

File: cmd/zstdseek/main.go [hunk 1]
```diff
@@ -60,13 +62,27 @@ func main() {
...
 	}
 
+	bar := progressbar.DefaultSilent(0, "")
+
 	var input io.ReadCloser
 	if inputFlag == "-" {
 		input = os.Stdin
 	} else {
-		if input, err = os.Open(inputFlag); err != nil {
+		file, err := os.Open(inputFlag)
```
Comment (SaveTheRbtz) [inline cmd/zstdseek/main.go:71]: nit: you do not need a separate `file` variable, you can use `input`.

File: cmd/zstdseek/main.go [hunk 2]
```diff
@@ -60,13 +62,27 @@ func main() {
...
+		if err != nil {
 			logger.Fatal("failed to open input", zap.Error(err))
 		}
+
+		if progressFlag {
+			stat, _ := file.Stat()
+
+			bar = progressbar.DefaultBytes(
+				stat.Size(),
+				"importing",
```
Comment (SaveTheRbtz) [inline cmd/zstdseek/main.go:81]: nit: compressing

File: cmd/zstdseek/main.go [hunk 3]
```diff
@@ -60,13 +62,27 @@ func main() {
...
+		file, err := os.Open(inputFlag)
+		if err != nil {
 			logger.Fatal("failed to open input", zap.Error(err))
 		}
+
+		if progressFlag {
+			stat, _ := file.Stat()
+
+			bar = progressbar.DefaultBytes(
+				stat.Size(),
```
Comment (SaveTheRbtz) [inline cmd/zstdseek/main.go:80]: you'll need to pass -1 to convert progressbar to spinner.  But if `file.Stat()` fails, it will return 0 as `.Size()`.

**Review by SaveTheRbtz — COMMENTED** [review]
please note that the project is using bazel, so you'll need to re-run [gazelle](https://github.com/bazelbuild/bazel-gazelle) and update deps, otherwise build/test will fail.

File: cmd/zstdseek/main.go (hunk 1 above)
Comment (jtarchie) [reply cmd/zstdseek/main.go:71]: The type for `input` is a `io.ReadCloser`, not the file. I can just use `os.Stat`, too.

**Comment by jtarchie** [comment]
First time using gazelle, I ran `gazelle update`. It seems to have made a delta related to the progress bar.

**Review by SaveTheRbtz — APPROVED** [review]

**Comment by SaveTheRbtz** [comment]
> First time using gazelle, I ran `gazelle update`. It seems to have made a delta related to the progress bar.

you'll also need to do `update-repos`.

**Comment by jtarchie** [comment]
I think I got it.

**Pull request closed without merging**